### PR TITLE
[🗺️ World Conquest⚔️] Renamed the Difficulty Levels to reflect their true Difficulties 

### LINKED ISSUE #7661: [🗺️ World Conquest⚔️] Rename 📝 the Difficulty Levels to their true 🔢 Names/Difficulties
### Context
The current difficulties start from Beginner to Expert. This is not true since Beginner level is actually equivalent to Normal and so on so the naming should be bumped by two levels.
Another issue is that the current existing naming is using both the old difficulty naming and the new naming conventions. This is confusing.

| Field | 1 | 2 | 3 | 4 | 5 | 6 |
| - | - | - | - | - | - | - |
|Old Name| Beginner | Easy | Medium | Hard | Challenging | Expert
|New Proposed Name| Normal | Challenging | Difficult | Expert |   Nightmare | Tormenting |

### PR DESCRIPTION
**Pull request by AugustVesterbacka**
resolves #7661

Implements @knyghtmare 's latest suggestion from issue #7661

Field | 1 | 2 | 3 | 4 | 5 | 6
-- | -- | -- | -- | -- | -- | --
Old Name | Beginner | Easy | Medium | Hard | Challenging | Expert
New Implemented Name | Normal | Challenging | Difficult | Fanatical | Torment | Nightmare
Unit Image | Peasant | Sergeant | Lieutenant | General | Marshal | Dragon

### CODE REVIEW

**Comment by Hejnewar** [comment]
One guy just asked us to make WC harder. That means nightmare isn't that bad. Also idk if fanatical or nightmare really are fine before or after nightmare. Kinda sound weird especially fanatical.

**Comment by soliton-** [comment]
What are "true Difficulties" based on? This does not seem like an improvement.

**Comment by gfgtdf** [comment]
> What are "true Difficulties" based on? This does not seem like an improvement.

The idea behind this is to make the difficulties comparable to the other normal (Singleplayer) campaigns, which in theory could be a good idea.

I personally cannot really judge on its difficulty compared to other campaign though, especially since when I play "usual" campaigns I just restart a scenario, while in wc2 that's rather uncommon from what I have observed when observing mp games. Afaik wc2 s difficulty is really designed around player who play the whole campaign without a single reload (even start of scenario saves)


> One guy just asked us to make WC harder. That means nightmare isn't that bad.

I'm not so sure about it, if I understood him correctly this is at least partly because he wants to play wc2 with mods (that usually make the game easier) and still have a challenge. In that case I think it should be those mods responsibility to make it harder.l, or at least to give an option for that.

> Also idk if fanatical or nightmare really are fine before or after nightmare. Kinda sound weird especially fanatical.

I agree that the new names are less obvious in their order than the previous ones.



So in conclusion I see the problem that this intends to solve, but I really don't think we should rush this into the 1.18 string freeze.

**Comment by gfgtdf** [comment]
At this point I want to quote the original author on the wc2 difficulty:


> I really want remark this issue to avoid players unnecessary frustation:
- Peasant is not a tutorial. WC II is a multiplayer campaign aimed to players with knowledge of BfW mechanics and Default units stats. This difficulty is a very difficult challenge for players that have just barely reach to that status.

- Liutenant is not a difficulty for noobs. It is a difficult challenge for players with average skill and a few years of experience in multiplayer server.

- Nightmare is not difficulty for average players. It is a very difficult challenge for players with high skill and lot of experience in World Conquest II.

> Challenge: I am a quite skilled player with high rating in ladder 1v1 and good results in 2v2 tournaments (sorry for lack of humilty, I thought was relevant information :oops:). I loved classic World Conquest and my main task was tune its difficulty to allow AI beat me (and 2 mates of similar skill) about 50% of times in Nightmare difficulty. I also did a great effort in redefine lower difficulty levels to just give more room players mistakes instead weak AI to a not challenging level. In other words, every player can find the right level of difficulty for his skill level to have a nice challenge. (completly begginers to Wesnoth will find this campaign probably too difficult and its mechanics too complex, so it is not recomended for them). Please, consulte "Warning about difficulty!:" spoiler in 1st post. (Note: WC is designed around replayability and difficulty is designed in asumption players will start a new campaign if they become defeated and save/reload is not used to cheat randomess. A single scenario can sometimes look not very challenge, but challenge is beat campaign.)

**Comment by knyghtmare** [comment]
> Peasant is not a tutorial. WC II is a multiplayer campaign aimed to players with knowledge of BfW mechanics and Default units stats. This difficulty is a very difficult challenge for players that have just barely reach to that status.

All the more reason to rename from "beginner" to "Normal". We are falsely advertising that the lowest difficulty of WC is winnable for players who are just starting out when it clearly is not.

I do think "Fanatical" should be renamed to something else which is synonymous, it just stands out too much.

>  if I understood him correctly this is at least partly because he wants to play wc2 with mods

There was also a request to add 1 easier difficulty for WC aimed at Casual players.

Lastly, I think renaming difficulties partially solves the real issue: WC needs to have 1-2 new difficulties aimed at beginners and/or casuals.

**Comment by Hejnewar** [comment]
> In that case I think it should be those mods responsibility to make it harder.l, or at least to give an option for that.

After a bit of talk thats indeed the case.

File: data/campaigns/World_Conquest/scenarios/WC_II_scenario.cfg [hunk 1]
```diff
@@ -102,12 +102,12 @@ _ "World Conquest 4p" #enddef
...
-        {WC2_CAMPAIGN_DIFFICULTY HARD {WC2_HUMAN_DIFFICULTY human-loyalists/general orange} _"General" _"Hard" 8 5 2 2 no 13}
-        {WC2_CAMPAIGN_DIFFICULTY VERY_HARD {WC2_HUMAN_DIFFICULTY human-loyalists/marshal white} _"Grand Marshal" _"Challenging" 9 6 2 1 no 17}
-        {WC2_CAMPAIGN_DIFFICULTY NIGHTMARE {WC2_NIGHTMARE_DIFFICULTY} _"Nightmare" _"Expert" 9 7 1 0 no 20}
+        {WC2_CAMPAIGN_DIFFICULTY VERY_EASY {WC2_HUMAN_DIFFICULTY human-peasants/peasant purple} _"Peasant" _"Normal" 6 2 2 10 yes 0}
+        {WC2_CAMPAIGN_DIFFICULTY EASY {WC2_HUMAN_DIFFICULTY human-loyalists/sergeant black} _"Sergeant" _"Challenging" 7 3 2 7 yes 5}
+        {WC2_CAMPAIGN_DIFFICULTY NORMAL {WC2_HUMAN_DIFFICULTY human-loyalists/lieutenant brown} _"Lieutenant" _"Difficult" 8 4 2 5 yes 10} {DEFAULT_DIFFICULTY}
+        {WC2_CAMPAIGN_DIFFICULTY HARD {WC2_HUMAN_DIFFICULTY human-loyalists/general orange} _"General" _"Fanatical" 8 5 2 2 no 13}
+        {WC2_CAMPAIGN_DIFFICULTY VERY_HARD {WC2_HUMAN_DIFFICULTY human-loyalists/marshal white} _"Grand Marshal" _"Torment" 9 6 2 1 no 17}
```
Comment (stevecotton) [inline data/campaigns/World_Conquest/scenarios/WC_II_scenario.cfg:109]: "Fanatical" and "Torment" are new strings. While no-one has taken the decision to formally say "no" to this PR, now that the string freeze has started I think it's time to say that no-one's approved it for 1.18, and that the window for merging it into 1.18 has closed.

Comment (knyghtmare) [reply data/campaigns/World_Conquest/scenarios/WC_II_scenario.cfg:109]: I am personally not happy with "Torment" and "Fanatical" terms being used here. I suggest closing this until we actually find suitable names for the difficulties. Also, 2 more easier and 1 more harder difficulty might be coming soon.
